app/screens/more_channels/index.js

```diff
@@ -29,20 +29,30 @@ const joinableChannels = createSelector(
     }
 );
 
+const teamArchivedChannels = createSelector(
+    getChannelsInCurrentTeam,
+    (channels) => {
+        return channels.filter((c) => c.delete_at !== 0);
+    }
+);
+
 function mapStateToProps(state) {
     const config = getConfig(state);
     const license = getLicense(state);
     const roles = getCurrentUserRoles(state);
-    const channels = joinableChannels(state);
+    const channels = joinablePublicChannels(state);
+    const archivedChannels = teamArchivedChannels(state);
     const currentTeamId = getCurrentTeamId(state);
 
     return {
         canCreateChannels: showCreateOption(state, config, license, currentTeamId, General.OPEN_CHANNEL, isAdmin(roles), isSystemAdmin(roles)),
         currentUserId: getCurrentUserId(state),
         currentTeamId,
         channels,
+        archivedChannels,
         theme: getTheme(state),
         isLandscape: isLandscape(state),
+        serverVersion: state.entities.general.serverVersion,
```

Comment:
I would apply the logic for the minimumServerVersion here and set a property like `canShowArchiveChannels` or something similar